File: src/dataset/go/bob_Exercism_go_ThomasZumsteg/claude/ClaudeSonnet4_bob_v4.go
package bob

import (
	"strings"
	"unicode"
)

const TestVersion = 1

func Hey(greeting string) string {
	greeting = strings.TrimSpace(greeting)

	if greeting == "" {
		return "Fine. Be that way!"
	}

	if greeting[len(greeting)-1] == '?' {
		return "Sure."
	}

	hasUpper := false
	hasLower := false
	hasLetter := false

	for _, r := range greeting {
		if unicode.IsLetter(r) {
			hasLetter = true
			if unicode.IsUpper(r) {
				hasUpper = true
			} else {
				hasLower = true
			}
			if hasUpper && hasLower {
				break
			}
		}
	}

	if hasLetter && hasUpper && !hasLower {
		return "Whoa, chill out!"
	}

	return "Whatever."
}

func any(items string, test func(rune) bool) bool {
	for _, item := range items {
		if test(item) {
			return true
		}
	}
	return false
}

File: src/dataset/typescript/bob_Exercism-typescript-shybyte/claude/ClaudeSonnet4_bob_v4.ts
class Bob {
    hey(messageRaw: string) {
        const message = messageRaw.trim()
        
        if (message.length === 0) {
            return 'Fine. Be that way!'
        }
        
        const hasAlpha = /[a-zA-Z]/.test(message)
        if (hasAlpha && message === message.toUpperCase()) {
            return 'Whoa, chill out!'
        }
        
        if (message.endsWith('?')) {
            return 'Sure.'
        }
        
        return 'Whatever.'
    }
}

export default Bob